### PR TITLE
Feat/arbitrary messages

### PR DESCRIPTION
**Pull request by JulianToledano**

### CODE REVIEW

File: cips/cip-offchain_message_signature.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+This CIP defines two different messages. The first one allows users to sign any arbitrary data, including files, text,
+objects, and more. The second one provides a way for application sign-in by defining a standard human-readable message
+format that is parameterized by scope, session details, and security mechanisms.
+
+## Motivation
+
+Having the ability to sign messages off-chain has proven to be a fundamental aspect of nearly any blockchain.
+The notion of signing messages off-chain has many added benefits, such as saving on computational costs and reducing
+transaction throughput and overhead. Within the context of the Cosmos, some of the major applications of signing such
```
Comment (raynaudoe) [inline cips/cip-offchain_message_signature.md]:
```suggestion
transaction throughput and overhead. Within the context of the Cosmos ecosystem, some of the major applications of signing such
```
Just an example

File: cips/cip-offchain_message_signature.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  "issued-at": "2023-10-10T10:10:10Z"
+```
+
+This approach is an improvement over the current state, as it allows for the signing of arbitrary data like files and
+offers a more user-friendly sign-in process, replacing non-human-readable random challenges.
+
+## Specification
+
+Off-chain signed messages should resemble Cosmos SDK messages but **must not** constitute a valid on-chain transaction.
+`chain-id`, `account_number`, and `sequence` must all be assigned invalid values.
```
Comment (raynaudoe) [inline cips/cip-offchain_message_signature.md:70]: Nothing serious, just a thing to keep in mind, I don't know the details about how the ledger app works but the app shouldn't reject the tx if any of those fields contains "Invalid" values.

File: cips/cip-offchain_message_signature.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+```protobuf
+// MsgSignArbitraryData defines an arbitrary, general-purpose, off-chain message
+message MsgSignArbitraryData {
+  option (cosmos.msg.v1.signer) = "signer";
+  // AppDomain is the application requesting off-chain message signing
+  string appDomain = 1;
+  // Signer is the sdk.AccAddress of the message signer
+  string signer = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Data represents the raw bytes of the content that is signed (text, json, etc)
+  bytes data = 3 [(gogoproto.jsontag) = "data"];
```
Comment (raynaudoe) [inline cips/cip-offchain_message_signature.md:112]: is there a size limit for `data` ? or maybe signing the digest of a file ?

**Review by raynaudoe — COMMENTED** [review]
Looks good, left a few comments

File: cips/cip-offchain_message_signature.md (hunk 2 above)
Comment (JulianToledano) [reply cips/cip-offchain_message_signature.md:70]: yeah, this has been tested and its not a problem.

File: cips/cip-offchain_message_signature.md (hunk 3 above)
Comment (JulianToledano) [reply cips/cip-offchain_message_signature.md:112]: good question, I guess the only limit here would come from HSM devices. Maybe its a good open question to add [here](https://github.com/Zondax/cosmos-cips/pull/1/files#diff-b00f3fdb788632ea03619fd3a122b48af64a7e96e433aa4af7912e2da90c0446R249)

File: cips/cip-offchain_message_signature.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+Proto definition:
+
+```protobuf
+// MsgSignArbitraryData defines an arbitrary, general-purpose, off-chain message
+message MsgSignArbitraryData {
+  option (cosmos.msg.v1.signer) = "signer";
+  // AppDomain is the application requesting off-chain message signing
+  string appDomain = 1;
+  // Signer is the sdk.AccAddress of the message signer
+  string signer = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```
Comment (bizk) [inline cips/cip-offchain_message_signature.md]: I think it would be better to clarify that the signer is an address, like `signerAddress`

**Review by bizk — APPROVED** [review]

**Pull request merged**